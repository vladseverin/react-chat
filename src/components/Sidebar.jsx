import React from 'react';
import { withStyles } from 'material-ui/styles';
import Drawer from 'material-ui/Drawer';
import Divider from 'material-ui/Divider';
import TextField from 'material-ui/TextField';
import List, { ListItem, ListItemText } from 'material-ui/List';
import BottomNavigation, { BottomNavigationAction } from 'material-ui/BottomNavigation';
import RestoreIcon from '@material-ui/icons/Restore';
import ExploreIcon from '@material-ui/icons/Explore';
import AddIcon from '@material-ui/icons/Add';
import Avatar from 'material-ui/Avatar';
import titleInitials from '../utils/title-initials';
import Button from 'material-ui/Button';

const styles = theme => ({
  drawerPaper: {
    overflow: 'hidden',
    position: 'relative',
    width: 320,
  },
  draverHeader: {
    ...theme.mixins.toolbar,
    paddingLeft: theme.spacing.unit * 3,
    paddingRight: theme.spacing.unit * 3,
  },
  input: {
    width: '100%',
    marginTop: 16,
    marginBottom: 8
  },
  chatsList: {
    height: 'calc(100% - 56px)',
    overflowY: 'scroll',
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  newChatButton: {
    position: 'absolute',
    left: 'auto',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3 + 48,
  },
  colorAvatar: {
    color: '#fff',
    backgroundColor: '#F44336',
  },
});

const Sidebar = ({ classes, chats }) => (
  <Drawer
    variant="permanent"
    classes={{
      paper: classes.drawerPaper,
    }}
  >
    {/* Implemented search bar */}
    <div className={classes.draverHeader} >
      <TextField
        fullWidth
        margin="normal"
        placeholder="Search chat..."
      />
    </div>
    <Divider />
    {/* Implemented list of chats */}
    <List className={classes.chatsList}>
      {chats.map((chat, index) => (
        <ListItem key={index} button>
          <Avatar className={classes.colorAvatar}>
            {
              titleInitials(chat.title)
            }
          </Avatar>
          <ListItemText primary={chat.title} secondary="Jan 9, 2014" />
        </ListItem>
      ))}
    </List>
    {/* Implemented button of create new chat */}
    <Button variant="fab" color="primary" aria-label="add" className={classes.newChatButton}>
      <AddIcon />
    </Button>
    {/* Implemented bottom navigation */}
    <BottomNavigation
      showLabels
    >
      <BottomNavigationAction label="My Chats" icon={<RestoreIcon />} />
      <BottomNavigationAction label="Explore" icon={<ExploreIcon />} />
    </BottomNavigation>
  </Drawer>
);

export default withStyles(styles)(Sidebar);
