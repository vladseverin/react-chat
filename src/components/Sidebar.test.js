import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './Sidebar';

jest.mock('./ChatList', () => () => 'ChatList');
jest.mock('./NewChatButton', () => () => 'NewChatButton');

const mockProps = {
  chats: {
    active: {},
    my: [],
    all: [],
  },
  createChat: jest.fn(),
  isConnected: true,
};

it('render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Sidebar {...mockProps} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
