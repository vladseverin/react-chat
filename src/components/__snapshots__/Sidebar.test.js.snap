// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<div
  className="MuiDrawer-docked-88"
>
  <div
    className="MuiPaper-root-99 MuiPaper-elevation0-101 MuiDrawer-paper-89 Sidebar-drawerPaper-85 MuiDrawer-paperAnchorLeft-90 MuiDrawer-paperAnchorDockedLeft-94"
  >
    <div
      className="Sidebar-drawerHeader-86"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-126 MuiFormControl-marginNormal-127 MuiFormControl-fullWidth-129"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiInput-root-130 MuiInput-fullWidth-137 MuiInput-formControl-131 MuiInput-underline-134"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-138"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder="Search chat..."
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <hr
      className="MuiDivider-root-143"
    />
    ChatList
    NewChatButton
    <div
      className="MuiBottomNavigation-root-147"
    >
      <button
        className="MuiButtonBase-root-153 MuiBottomNavigationAction-root-148 MuiBottomNavigationAction-selected-149"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiBottomNavigationAction-wrapper-151"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-156"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
              />
            </g>
          </svg>
          <span
            className="MuiBottomNavigationAction-label-152 MuiBottomNavigationAction-selected-149"
          >
            My Chats
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-162"
        />
      </button>
      <button
        className="MuiButtonBase-root-153 MuiBottomNavigationAction-root-148"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiBottomNavigationAction-wrapper-151"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-156"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
              />
            </g>
          </svg>
          <span
            className="MuiBottomNavigationAction-label-152"
          >
            Explore
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-162"
        />
      </button>
    </div>
  </div>
</div>
`;
